SynonymGroup.ts

```diff
@@ -1,192 +1,736 @@
+      query,
+      { signal: this.controller.signal },
+      `Subtaxa ${url}`,
+    );
+
+    const names = json.results.bindings
+      .map((n) => n.sub?.value)
+      .filter((n) => n && !this.expanded.has(n)) as string[];
+
+    await Promise.allSettled(
+      names.map((n) => this.getName(n, { searchTerm: true, subTaxon: true })),
+    );
+  }
+
+  /** @internal */
+  private async getNameFromLatin(
+    [genus, species, infrasp]: [string, string | undefined, string | undefined],
+    justification: Justification,
+  ): Promise<void> {
+    const query = `
+    PREFIX dwc: <http://rs.tdwg.org/dwc/terms/>
+SELECT DISTINCT ?uri WHERE {
+  ?uri dwc:genus|dwc:genericName "${genus}" .
+  ${
+      species
+        ? `?uri dwc:species|dwc:specificEpithet "${species}" .`
+        : "FILTER NOT EXISTS { ?uri dwc:species|dwc:specificEpithet ?species . }"
+    }
+  ${
+      infrasp
+        ? `?uri dwc:subspecies|dwc:variety|dwc:infraspecificEpithet "${infrasp}" .`
+        : "FILTER NOT EXISTS { ?uri dwc:subspecies|dwc:variety|dwc:infraspecificEpithet ?infrasp . }"
+    }
+}
+LIMIT 500`;
+
+    if (this.controller.signal?.aborted) return Promise.reject();
+    const json = await this.sparqlEndpoint.getSparqlResultSet(
+      query,
+      { signal: this.controller.signal },
+      `NameFromLatin ${genus} ${species} ${infrasp}`,
+    );
+
+    const names = json.results.bindings
+      .map((n) => n.uri?.value)
+      .filter((n) => n && !this.expanded.has(n)) as string[];
+
+    await Promise.allSettled(names.map((n) => this.getName(n, justification)));
+  }
+
+  /** @internal */
+  private async getNameFromCol(
+    colUri: string,
+    justification: Justification,
+  ): Promise<void> {
+    // Note: this query assumes that there is no sub-species taxa with missing dwc:species
+    // Note: the handling assumes that at most one taxon-name matches this colTaxon
+    const query = `
+PREFIX dwc: <http://rs.tdwg.org/dwc/terms/>
+PREFIX dwcFP: <http://filteredpush.org/ontologies/oa/dwcFP#>
+PREFIX cito: <http://purl.org/spar/cito/>
+PREFIX trt: <http://plazi.org/vocab/treatment#>
+SELECT DISTINCT ?tn ?tc ?col ?rank ?genus ?species ?infrasp ?name ?authority
+  (group_concat(DISTINCT ?tcauth;separator=" / ") AS ?tcAuth)
+  (group_concat(DISTINCT ?aug;separator="|") as ?augs)
+  (group_concat(DISTINCT ?def;separator="|") as ?defs)
+  (group_concat(DISTINCT ?dpr;separator="|") as ?dprs)
+  (group_concat(DISTINCT ?cite;separator="|") as ?cites)
+  (group_concat(DISTINCT ?trtn;separator="|") as ?tntreats)
+  (group_concat(DISTINCT ?citetn;separator="|") as ?tncites) WHERE {
+  BIND(<${colUri}> as ?col)
+  ?col dwc:taxonRank ?rank .
+  OPTIONAL { ?col dwc:scientificNameAuthorship ?colAuth . } BIND(COALESCE(?colAuth, "") as ?authority)
+  ?col dwc:scientificName ?name . # Note: contains authority
+  ?col dwc:genericName ?genus .
+  # TODO # ?col dwc:parent* ?p . ?p dwc:rank "kingdom" ; dwc:taxonName ?kingdom .
```

Comment:
This (and the other similar "# TODO #" lines are a hack to work around lindas not having all CoL-taxa for some reason?

Should be uncommented once all col-data are on lindas; this check prevents some weird edgecases like for example with "Quercus"